File: C_training/training/C/mephi/Stack/stack.c
#include "stack.h"

void ft_push(int item, t_stack *obj)
{
	if (obj->tos > obj->size_stack)
	{
		printf("error stack overflow\n");
		return ;
	}
	else
		obj->num[++obj->tos] = item;
}

int	ft_pop(t_stack *obj)
{
	if (obj->tos < 0)
	{
		printf("stack empty\n");
		return (0);
	}
	else
		return (obj->num[obj->tos--]);
}

File: C_training/training/java/java_test/src/Stack.java
public class Stack
{
	private int stck[];
	private int tos;
	public	int size;

	Stack(int num)
	{
		this.size = num;
		stck = new int[num];
		tos = -1;
	}

	public void push(int num)
	{
		if (tos == stck.length - 1)
		{
			System.out.println("Стэк переполнен");
		}
		else
			stck[++tos] = num;
	}

	public int pop()
	{
		if (tos < 0)
		{
			System.out.println("Стэк пуст");
			return (0);
		}
		else
			return (stck[tos--]);
	}
}